src/main/java/seedu/intrack/logic/commands/AddTagCommand.java

```diff
@@ -0,0 +1,89 @@
+package seedu.intrack.logic.commands;
+
+import static seedu.intrack.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.intrack.model.Model.PREDICATE_SHOW_ALL_INTERNSHIPS;
+
+import java.util.List;
+import java.util.Set;
+
+import seedu.intrack.commons.core.Messages;
+import seedu.intrack.commons.core.index.Index;
+import seedu.intrack.logic.commands.exceptions.CommandException;
+import seedu.intrack.model.Model;
+import seedu.intrack.model.internship.Internship;
+import seedu.intrack.model.tag.Tag;
+
+/**
+ * Adds a new Tag to the selected Internship.
+ */
+public class AddTagCommand extends Command {
+    public static final String COMMAND_WORD = "addtag";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds one or more tags to the internship application "
+            + "with the " + "selected index.\n" + "Parameters: INDEX (must be a positive integer) TAG \n"
+            + "Example: " + COMMAND_WORD + " 1 URGENT\n"
+            + "or " + COMMAND_WORD + " 1 URGENT HARD HELP";
+
+    public static final String MESSAGE_ADD_TAG_SUCCESS = "Added tag to internship successfully";
```

Comment:
Maybe you can consider targeting the success message to specify which internship has been changed?

Reply:
Added it by overriding toString() in Index class so that Index of selected Internship can be reflected in message.